api/jobset/v1alpha2/jobset_webhook.go

```diff
@@ -73,6 +74,14 @@ func (js *JobSet) ValidateCreate() error {
 	var allErrs []error
 	// Validate that replicatedJobs listed in success policy are part of this JobSet.
 	validReplicatedJobs := replicatedJobNamesFromSpec(js)
+	for _, rjob := range js.Spec.ReplicatedJobs {
+		if rjob.Template.Spec.Parallelism == nil {
+			rjob.Template.Spec.Parallelism = pointer.Int32(1)
```

Comment:
we are not supposed to change the value, just use a temp variable named `parallelism`

```
parallelism = 1
if rjob.Template.Spec.Parallelism != nil {
  parallelism = *rjob.Template.Spec.Parallelism
}
```